src/algebra/group_action_hom.lean

```diff
@@ -255,4 +255,33 @@ ext $ λ x, by rw [comp_apply, id_apply]
 @[simp] lemma comp_id (f : R →+*[M] S) : f.comp (mul_semiring_action_hom.id M) = f :=
 ext $ λ x, by rw [comp_apply, id_apply]
 
+variables {P : Type*} [comm_semiring P] [mul_semiring_action M P]
+variables {Q : Type*} [comm_semiring Q] [mul_semiring_action M Q]
+open polynomial
+
+/-- An equivariant map induces an equivariant map on polynomials. -/
+protected noncomputable def polynomial (g : P →+*[M] Q) : polynomial P →+*[M] polynomial Q :=
+⟨map g, λ m p, polynomial.induction_on p
```

Comment:
I personally prefer the syntax
```
{ to_fun := _,
  map_add' := _ } -- etc
```
because it gives the casual reader a lot more sense for what's being proven.

Reply:
done